src/proxy.rs

```diff
@@ -469,6 +477,102 @@ impl Dst for Uri {
     }
 }
 
+/// Get system proxies information.
+///
+/// It can only support Linux, Unix like, and windows system.  Note that it will always
+/// return a HashMap, even if something runs into error when find registry information in
+/// Windows system.  Note that invalid proxy url in the system setting will be ignored.
+///
+/// Returns:
+///     System proxies information as a hashmap like
+///     {"http": Url::parse("http://127.0.0.1:80"), "https": Url::parse("https://127.0.0.1:80")}
+pub fn get_proxies() -> HashMap<String, Url> {
+    let proxies: HashMap<String, Url> = get_from_environment();
+
+    if proxies.len() == 0 {
+        // don't care errors if can't get proxies from registry, just return an empty HashMap.
+        #[cfg(target_os = "windows")]
+        return get_from_registry();
+    }
+    return proxies;
```

Comment:
```suggestion
    proxies
```
`return` shouldn't be needed